cdk/stack/ynr.py

```diff
@@ -111,6 +119,16 @@ def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
             "YNR_DJANGO_SECRET_KEY": "insecure",
         }
 
+        # `alb_basic_auth_token` prevents anyone from accessing the ALB without
+        # passing this header. We use it to limit hosts to valid hostnames
+        # and to ensure no one can access the ALB directly.
+
+        self.alb_basic_auth_token = (
+            ssm.StringParameter.value_for_string_parameter(
+                self, "alb_basic_auth_token"
+            )
+        )
```

Comment:
One of the things Matt did on this repo (which we haven't done on other repos) is set some params as string and some as secure string. So `YNR_AWS_S3_MEDIA_REGION` is just a string but `POSTGRES_PASSWORD` is encrypted. Given that, I reckon `alb_basic_auth_token` should be secure string using `from_secure_string_parameter_attributes()`